src/Flagged/Modal.jsx

```diff
@@ -0,0 +1,51 @@
+let {
+  onReport,
+  reportedAccountId,
+  contentModerationFlagValue,
+  ...forwardedProps
+} = props;
+
+const socialSet = (index) =>
+  Social.set(index, {
+    onCommit: () => {
+      onReport && onReport();
+    },
+    onCancel: () => onOpenChange(false),
+  });
+
+const reportAccount = () => {
+  socialSet({
+    index: {
+      flag: JSON.stringify({
+        key: "main",
+        value: contentModerationFlagValue,
+      }),
+    },
+  });
+};
+
+const reportAccountWithPosts = () => {
+  socialSet({
+    index: {
+      moderate: {
+        [reportedAccountId]: "report",
+      },
+    },
```

Comment:
```suggestion
    index: {
      moderation: {
        key: "reported",
        value: {
          path: reportedAccountId,
          label: "report"
        },
        [reportedAccountId]: "report",
      },
    },
    moderation: {
      [reportedAccountId]: "report",
    },
```


@gabehamilton according to your comment below is this what you meant?

Reply:
The index.moderation value needs to be a string for the index standard.
```suggestion
      index: {
          moderation: `{"key":"reported","value":{"path":"${reportedAccountId}", "label": "report"}}`
      },
      moderation: {
          [reportedAccountId]: "report",
      },

```

Reply:
Thanks @gabehamilton!